rfc/text/advanced/ap_48_uris.md

```diff
@@ -2,6 +2,62 @@
 
 WAMP pre-defines the following error URIs for the **Advanced Profile**. WAMP peers SHOULD only use the defined error messages.
 
+## Authentication
+
+No authentication method the *Client* offered is accepted.
+
+{align="left"}
+        wamp.error.no_auth_method
+
+The *Client* attempted to authenticate for a *Realm* (`realm|string`) that does not (or no longer) exists.
+
+{align="left"}
+        wamp.error.no_such_realm
+
+The *Client* attempted to authenticate for a *Role* (`authrole|string`) that does not (or no longer) exists.
+
+{align="left"}
+        wamp.error.no_such_role
+
+The *Client* authenticated for a *Principal* (`authid|string`) that does not (or no longer) exists.
+
+{align="left"}
+        wamp.error.no_such_principal
+
+The authentication as presented by the *Client* is denied (e.g. "wrong password").
+
+{align="left"}
+        wamp.error.authentication_denied
+
+The authentication of the *Client* failed technically at run-time, and the *Client* is therefor rejected ("fail secure operation").
+
+{align="left"}
+        wamp.error.authentication_failed
+
+The *Client* did not authenticate (at all) and a successful (non-anonymous) authentication is required.
```

Comment:
```suggestion
The *Client* did not provide the required, non-anonymous, authentication information.
```

Reply:
Referring to success here is confusing, because there are URI devoted to that: `wamp.error.authentication_denied` and `wamp.error.authentication_failed`.